### DISCUSSION TITLE
Set votes needed to get a rating close to zero

### DISCUSSION THREAD

**Discussion by local-minimum** [ludumdare/ludumdare#2018] [General]
If we change how votes are calculated from a simple average to something that takes uncertainty into account, simple and transparent  solution would be to add `N` angry voters that give you a `1` each were `N` is the same for everyone independent of how many votes they really had. Suitable `N` can easily be tested.

```php
rating = (Sum($votes) + $N) / (Count($votes) + $N)
```

`$N=3` or there about is probably enough to allow people to get rating at even less than 5.

One bad thing is that you can't directly compare to your previous ratings if you have very few votes.

**Comment by local-minimum** [comment]
If everyone gets one angry voter:
![votes_n1](https://cloud.githubusercontent.com/assets/8041100/26278624/7a99cfd8-3d9f-11e7-982b-9906cde77f71.png)

If everyone gets two angry voters:
![votes_n2](https://cloud.githubusercontent.com/assets/8041100/26278623/7a94c150-3d9f-11e7-8d73-b6f36c5f257f.png)

Three and you really need a lot of votes to get anywhere near 5.0 in end score, but perhaps that is OK? Anyhow the damping effect on ppl with few votes on their game is already strong with 2 angry voters.

**Comment by local-minimum** [comment]
For reference, N=3 angry voters.

![votes_n3](https://cloud.githubusercontent.com/assets/8041100/26278717/69c22582-3da1-11e7-9c51-30c8bdaaa350.png)

**Comment by zwrawr** [comment]
I don't think this would allow us to give users with low votes accurate ratings.
I.e. with 5 votes all 5 star this gives a rating of 3.5, that's way to different.
It's still 0.5 out when 20 people have rated the game.

**Comment by local-minimum** [comment]
It's not really intending on giving accurate ratings, it's intended to allow giving ratings to few voted games without the risk of the game being friend-voted to a too high place. It would also give a soft incentive to keep voting on games, especially if you think you have a winner. We could of course also indicate how much they lost due to having voted on too few games themselves.

**Comment by local-minimum** [comment]
Alternative and a bit more complex math to make the few-vote scaling more quickly come close to true average (however calculated) by letting number of angry voters diminish with number of games voted on.

Algorithm would be something like this:

```php
$score = (Sum($votes_on_game) + Max(0, $N - log(Count($my_votes_on_others) + 1)) / (Count($votes_on_game) + Max(0, $N - log(Count($my_votes_on_others) + 1));
```

Looking at the most extreme case, when a game has a true average of 5.0:

![votes_model2_target5](https://cloud.githubusercontent.com/assets/8041100/26279681/95bbc8f0-3dba-11e7-8264-9a6ea35119f3.png)

You can see it requires few games played and not that many votes received to get close to true rating. Many combinations here are extremely unlikely, e.g. a game that has played 20 games but only gotten one vote and it's a 5.

With other true averages:
![votes_model2_target2](https://cloud.githubusercontent.com/assets/8041100/26279713/2a0b8432-3dbb-11e7-8a93-fa88fc017446.png)
![votes_model2_target4](https://cloud.githubusercontent.com/assets/8041100/26279712/2a095a86-3dbb-11e7-968e-773a8550db32.png)

**Comment by local-minimum** [comment]
As you can see the plots look the same because the effect behaves the same, the magnitude of the effect is only larger for 5 compared to 4 compared to 2 (values-cale changes).

**Comment by local-minimum** [comment]
Using the category `Overall` and actual data from LD38, the proposed method using a smaller modification of the second proposed method  and `N=4.4` and using log2 rather than natural logarithm. This means that the adjustment effect becomes zero if you've voted on 20 or more games and the strength of the adjustment increases with the fewer than 20 games you've _voted on_ as well as with the higher average you got and the fewer votes you got.

I think this shows a few games would get an adjustment but most results would not be affected:

![ld38overalladjusted](https://cloud.githubusercontent.com/assets/8041100/26447786/69d077ac-414a-11e7-8625-481f46aa96ca.png)

Or looking at the size of the adjustments:

![ld38overalladjustment](https://cloud.githubusercontent.com/assets/8041100/26448427/7e22c7c0-414d-11e7-9925-ce6bb91a1d03.png)


This would have resulted in the following order for the top ten games (adjusted winner first), numbers are original rank:

`[5, 3, 4, 2, 8, 9, 14, 1, 10, 6]`

Most notably the winner would fall down a couple of placements, basically due to them not having been good participants and only playing 8 games. This is the intent of the proposed method. They still rank very high, which is good because the game probably deserved ending up close to the top.

There's one vulnerability in the method and that is if I have 3-4 votes that are all 5 because friends gave me these votes and then last minute I go in and vote on 20 games. Then my adjusted score would be 5 still. This is highly unlikely as far as I see it.

**Comment by mikekasprzak** [comment]
> If we change how votes are calculated from a simple average

It's not a simple average. Highest and lowest rating are discarded. I would discard more if I could reliably get enough votes for games. :wink:

**Comment by local-minimum** [comment]
yes but the methods work well together and my suggestion would probably mean you don't have to remove add many and therefore not loose add much information about how ppl perceived the game.

i think something like this could be good to foster participation in the voting because you could easily say to them how many more games needs playing before they are not punished for slacking.

but anyways I'm not saying weer must do this or anything.
